locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+        |                       |  Setup event  |
+        |                       |  closures:    |
++-------v-----------+           |               |         +-------------------+
+|  Readable Stream  >-----------+ * .data(cb)   +--------->  Writable Stream  |
++-^-------^-------^-+           | * .drain(cb)  |         +-------------------+
+  ^       ^       ^             | * .unpipe(cb) |                   |
+  |       ^       |             | * .error(cb)  |                   |
+  |       |       |             | * .finish(cb) |                   |
+  ^       |       |             | * .end(cb)    |           +=======v=========+
+  ^       |       ^             | * .close(cb)  |           |  .write(chunk)  |
+  |       |       ^             +---------------+           +=======+=========+
+  |       ^       |                                                 |
+  |       ^       |                              +------------------v---------+
+  ^       |       +-> if (!chunk)                |    Is this chunk too big?  |
+  ^       |       |     emit .end();             |    Is the queue busy?      |
+  |       |       +-> else                       +-------+----------------+---+
+  |       ^       |     emit .write();                   |                |
+  |       ^       ^                                   +--v---+        +---v---+
+  |       |       ^-----------------------------------<  No  |        |  Yes  |
+  ^       |                                           +------+        +---v---+
+  ^       |                                                               |
+  |       ^               emit .pause();          +=================+     |
+  |       ^---------------^-----------------------+  return false;  <-----+---+
+  |                                               +=================+         |
+  |                                                                           |
+  ^            when queue is empty     +============+                         |
+  ^------------^-----------------------<  Buffering |                         |
+               |                       |============|                         |
+               +> emit .drain();       |  ^Buffer^  |                         |
+               +> emit .resume();      +------------+                         |
+                                       |  ^Buffer^  |                         |
+                                       +------------+   add chunk to queue    |
+                                       |            <---^---------------------<
+                                       +============+
+```
+
+_Note:_ The `.pipe()` function is typically where backpressure is invoked. In
+an isolate application, both [`Readable`][] and [`Writable`][] streams
+should be present. If you are writing an application meant to accept a
+[`Readable`][] stream, or pipes to a [`Writable`][] stream from another
+component, you may omit this detail.
```

Comment:
So, these are the true facts say about a `Transform` stream:

It is important because its used often to setup pipelines. In it, there are `_write()` and `_read()` calls that must be implemented with the same care as with `Readable` and `Writable` (with respect to the return values `.push()` and `.write()`. And also HWM may change along the pipeline.